services/jobs/src/test/java/com/linkedin/openhouse/jobs/mock/JobsApiValidatorTest.java

```diff
@@ -44,4 +55,51 @@ public void validateJobRequestBody() {
             jobsApiValidator.validateCreateJob(
                 makeJobRequestBodyFromJobNameClusterId("nothing123", "crazy456")));
   }
+
+  @Test
+  public void testValidMemoryFormatInJobRequestBody() {
+    // Ensure hyphen is fine in clusterId and JobName
+    assertDoesNotThrow(
+        () ->
+            jobsApiValidator.validateCreateJob(
+                makeJobRequestBodyFromJobNameJobConf("job-name", "4G")));
+
+    assertDoesNotThrow(
+        () ->
+            jobsApiValidator.validateCreateJob(
+                makeJobRequestBodyFromJobNameJobConf("job-name", "10G")));
+
+    assertDoesNotThrow(
+        () ->
+            jobsApiValidator.validateCreateJob(
+                makeJobRequestBodyFromJobNameJobConf("job-name", "256M")));
+  }
+
+  @Test
+  public void testInValidMemoryFormatInJobRequestBody() {
+    // Ensure hyphen is fine in clusterId and JobName
+    assertThrows(
+        RequestValidationFailureException.class,
```

Comment:
Can we add a failure test for 10MG as well?

Reply:
Done